src/components/Header/Header.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import style from './Header.module.scss';
+import { MdLocalMovies } from 'react-icons/md';
+import { BsSearch } from 'react-icons/bs';
+
+interface HeaderProps {
+  value: string;
+  handleSearch(): Promise<void>;
+}
+
+class Header extends React.Component<HeaderProps, { value: string }> {
+  constructor(props: HeaderProps) {
+    super(props);
+    this.state = {
+      value: props.value || '',
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleButtonClick = this.handleButtonClick.bind(this);
+  }
+
+  componentDidMount() {
+    const savedValue = localStorage.getItem('TVShowSearch') || '';
+    this.setState({ value: savedValue });
+  }
+
+  handleChange(e: React.FormEvent<HTMLInputElement>) {
```

Comment:
Please, use more descriptive variable names - "event" instead of "e"

Reply:
Done